notebook/Dockerfile

```diff
@@ -1,37 +1,45 @@
-FROM jupyter/base-notebook
+FROM jupyter/base-notebook:4d7dd95017ed
```

Comment:
Why this specific hash?

Reply:
That was "latest" of the base image when I made my edits. If we don't specify a version, the base image may have newer versions than we're requesting, triggering a downgrade in some packages. 

This happened recently and caused some bloat in our image: the current base image contains `jupyterlab==0.35.5` but we were calling for `jupyterlab==0.35.4`, so it downloaded and installed the lower version.

Another way around this is would be less restrictive in specifying package versions: `jupyterlab>=0.35.4` should also avoid the downgrade.